internal/master/service.go

```diff
@@ -31,36 +33,77 @@ type DKVService interface {
 	serverpb.DKVDiscoveryNodeServer
 	serverpb.DKVReplicationServer
 	serverpb.DKVBackupRestoreServer
+	serverpb.HealthCheckServer
 }
 
 type standaloneService struct {
-	store storage.KVStore
-	cp    storage.ChangePropagator
-	br    storage.Backupable
-
+	store      storage.KVStore
+	cp         storage.ChangePropagator
+	br         storage.Backupable
 	rwl        *sync.RWMutex
-	lg         *zap.Logger
-	statsCli   stats.Client
 	regionInfo *serverpb.RegionInfo
+	isClosed   bool
+	shutdown   chan struct{}
+	opts       serveropts.ServerOpts
 }
 
 func (ss *standaloneService) GetStatus(ctx context.Context, request *emptypb.Empty) (*serverpb.RegionInfo, error) {
 	return ss.regionInfo, nil
 }
 
+func (ss *standaloneService) Check(ctx context.Context, healthCheckReq *serverpb.HealthCheckRequest) (*serverpb.HealthCheckResponse, error) {
+	if !ss.isClosed && ss.regionInfo != nil && ss.regionInfo.Status == serverpb.RegionStatus_LEADER {
+		return &serverpb.HealthCheckResponse{Status: serverpb.HealthCheckResponse_SERVING}, nil
+	}
+	return &serverpb.HealthCheckResponse{Status: serverpb.HealthCheckResponse_NOT_SERVING}, nil
+}
+
+func (ss *standaloneService) Watch(req *serverpb.HealthCheckRequest, watcher serverpb.HealthCheck_WatchServer) error {
+	// if the call is made after service shutdown, it should always return not serving health check
+	// response
+	if ss.isClosed {
+		if err := checkAndSendResponse(req, watcher, ss); err != nil {
+			return err
+		}
```

Comment:
missing return statement. Otherwise the section below will still be executed.

Reply:
Added below this.